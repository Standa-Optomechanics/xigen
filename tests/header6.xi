protocol "v1"
defaults with crc, answer, public

/** $XIS
Usual doxygen struct
*/
/** $XIW
Usual doxygen write
*/
/** $XIR
Usual doxygen read
*/
command "testdoc1" universal "doc" (10) 
fields:
	int32s foo

/** $XIW
Usual doxygen write
*/
/** $XIS
Usual doxygen struct
*/
/** $XIR
Usual doxygen read
*/
command "testdoc2" universal "doc" (10) 
fields:
	int32s foo

/** $XIR
Usual doxygen read
*/
/** $XIS
Usual doxygen struct
*/
command "testdoc3" reader "tdoc" (10) 
fields:
	int32s foo

/** $XIS
Usual doxygen struct
*/
/** $XIW
Usual doxygen write
*/
/** $XIR
Usual doxygen read
*/
/** $XISC
Usual doxygen struct calb
*/
/** $XIRC
Usual doxygen read calb
*/
/** $XIWC
Usual doxygen write calb
*/
command "testdoc4" universal "doc" (12) 
fields:
	calb float foo
	normal int32s foo
	normal int16s ufoo


