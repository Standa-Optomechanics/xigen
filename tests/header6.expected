
/** @cond DO_NOT_WANT */
#define LIBXIMC_PROTOCOL_VERSION 1
/** @endcond */


/*
 ------------------------------------------
   BEGIN OF GENERATED struct declarations
 ------------------------------------------
*/

/** 
Usual doxygen struct
	* @see get_testdoc1, set_testdoc1
*/
	typedef struct
	{
		int foo;
	} testdoc1_t;

/** 
Usual doxygen struct
	* @see get_testdoc2, set_testdoc2
*/
	typedef struct
	{
		int foo;
	} testdoc2_t;

/** 
Usual doxygen struct
	* @see get_testdoc3
*/
	typedef struct
	{
		int foo;
	} testdoc3_t;

/** 
Usual doxygen struct
	* @see get_testdoc4, set_testdoc4
*/
	typedef struct
	{
		int foo;
		int ufoo;
	} testdoc4_t;

/** 
Usual doxygen struct calb
	* @see get_testdoc4, set_testdoc4
*/
	typedef struct
	{
		float foo;
	} testdoc4_calb_t;

/*
 --------------------------------------------
   BEGIN OF GENERATED function declarations
 --------------------------------------------
*/
/** 
Usual doxygen write
*/
	result_t XIMC_API set_testdoc1 (device_t id, const testdoc1_t* testdoc1);

/** 
Usual doxygen read
*/
	result_t XIMC_API get_testdoc1 (device_t id, testdoc1_t* testdoc1);

/** 
Usual doxygen write
*/
	result_t XIMC_API set_testdoc2 (device_t id, const testdoc2_t* testdoc2);

/** 
Usual doxygen read
*/
	result_t XIMC_API get_testdoc2 (device_t id, testdoc2_t* testdoc2);

/** 
Usual doxygen read
*/
	result_t XIMC_API get_testdoc3 (device_t id, testdoc3_t* testdoc3);

/** 
Usual doxygen write
*/
	result_t XIMC_API set_testdoc4 (device_t id, const testdoc4_t* testdoc4);

/** 
Usual doxygen write calb
*/
	result_t XIMC_API set_testdoc4_calb (device_t id, const testdoc4_calb_t* testdoc4_calb, const calibration_t* calibration);

/** 
Usual doxygen read
*/
	result_t XIMC_API get_testdoc4 (device_t id, testdoc4_t* testdoc4);

/** 
Usual doxygen read calb
*/
	result_t XIMC_API get_testdoc4_calb (device_t id, testdoc4_calb_t* testdoc4_calb, const calibration_t* calibration);


/*
 -------------------------
   END OF GENERATED CODE
 -------------------------
*/
